comp/core/flare/flare.go

```diff
@@ -47,28 +48,43 @@ type dependencies struct {
 	AC                    optional.Option[autodiscovery.Component]
 }
 
-type flare struct {
+type provides struct {
+	fx.Out
+
+	Comp     Component
+	Endpoint api.AgentEndpointProvider
+}
+
+type Flare struct {
```

Comment:
Why did you make the flare struct public?

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/24045/commits/fb28afd4d8f659ee1ba4740eb558d82e2fa9715b